railgun/apis/configuration/v1beta1/ingress_rules.go

```diff
@@ -0,0 +1,44 @@
+package v1beta1
+
+// UDPIngressRule represents a rule to apply against incoming requests
+// wherein no Host matching is available for request routing, only the port
+// is used to match requests.
+type UDPIngressRule struct {
+	// Port indicates the port for the Kong proxy to accept incoming traffic
+	// on, which will then be routed to the service Backend.
+	Port int `json:"port"`
+
+	// Backend defines the Kubernetes service which accepts traffic from the
+	// listening Port defined above.
+	Backend IngressBackend `json:"backend"`
+}
+
+// IngressRule represents a rule to apply against incoming requests.
+// Matching is performed based on an (optional) SNI and port.
+type IngressRule struct {
+	// Host is the fully qualified domain name of a network host, as defined
+	// by RFC 3986.
+	// If a Host is specified, the protocol must be TLS over TCP.
+	// A plain-text TCP request cannot be routed based on Host. It can only
+	// be routed based on Port.
+	// +optional
+	Host string `json:"host,omitempty"`
+
+	// Port is the port on which to accept TCP or TLS over TCP sessions and
+	// route. It is a required field. If a Host is not specified, the requested
+	// are routed based only on Port.
+	Port int `json:"port,omitempty"`
```

Comment:
Do you think is a good idea to use "kubebuilder:validate:minPort=1:maxPort=65535" do the validation automatically ?

Reply:
Yes that's a great idea let's do that:

```suggestion
	// +kubebuilder:validation:Minimum=1
	// +kubebuilder:validation:Maximum=65535
	Port int `json:"port,omitempty"`
```